File: app/src/main/java/com/alastor/spacex/SpaceXService.java
package com.alastor.spacex;

import com.alastor.spacex.model.Capsule;
import com.alastor.spacex.model.Company;
import com.alastor.spacex.model.Core;
import com.alastor.spacex.model.Crew;
import com.alastor.spacex.model.Dragon;
import com.alastor.spacex.model.Landpad;
import com.alastor.spacex.model.Launch;
import com.alastor.spacex.model.Launchpad;
import com.alastor.spacex.model.Payload;
import com.alastor.spacex.model.Roadster;
import com.alastor.spacex.model.Rocket;
import com.alastor.spacex.model.Ship;
import com.alastor.spacex.model.Starlink;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface SpaceXService {

    @GET("/capsules")
    Single<List<Capsule>> getCapsules();

    @GET("/capsules/{id}")
    Single<Capsule> getCapsule(@Path("id") long id);

    @GET("/company")
    Single<Company> getCompanyInfo();

    @GET("/cores")
    Single<List<Core>> getCores();

    @GET("/cores/{id}")
    Single<Core> getCore(@Path("id") long id);

    @GET("/crew")
    Single<List<Crew>> getCrews();

    @GET("/crew/{id}")
    Single<Crew> getCrew(@Path("id") long id);

    @GET("/dragons")
    Single<List<Dragon>> getDragons();

    @GET("/dragons/{id}")
    Single<Dragon> getDragon(@Path("id") long id);

    @GET("/landpads")
    Single<List<Landpad>> getLandpads();

    @GET("/landpads/{id}")
    Single<Landpad> getLandpad(@Path("id") long id);

    @GET("/launches/past")
    Single<List<Launch>> getPastLaunches();

    @GET("/launches/upcoming")
    Single<Launch> getUpcomingLaunches();

    @GET("/launches/latest")
    Single<Launch> getLatestLaunch();

    @GET("/launches/next")
    Single<Launch> getNextLaunch();

    @GET("/launches")
    Single<List<Launch>> getAllLaunches();

    @GET("/launches/{id}")
    Single<Landpad> getLaunch(@Path("id") long id);

    @GET("/launchpads")
    Single<List<Launchpad>> getLaunchPads();

    @GET("/launchpads/{id}")
    Single<Launchpad> getLaunchPad(@Path("id") long id);

    @GET("/payloads")
    Single<List<Payload>> getPayLoads();

    @GET("/payloads/{id}")
    Single<Payload> getPayLoad(@Path("id") long id);

    @GET("/roadster")
    Single<List<Roadster>> getRoadster();

    @GET("/rockets")
    Single<List<Rocket>> getRockets();

    @GET("/rockets/{id}")
    Single<Rocket> getRocket(@Path("id") long id);

    @GET("/ships")
    Single<List<Ship>> getShips();

    @GET("/ships/{id}")
    Single<Ship> getShip(@Path("id") long id);

    @GET("/starlink")
    Single<List<Starlink>> getStarlinks();

    @GET("/starlink/{id}")
    Single<Starlink> getStarlink(@Path("id") long id);

}

File: app/src/main/java/com/alastor/spacex/SpaceXService.kt
package com.alastor.spacex

import com.alastor.spacex.model.*
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface SpaceXService {

    @GET("/company")
    fun getCompanyInfo(): Single<Company>

    @GET("/cores")
    fun getCores(): Single<List<Core>>

    @GET("/cores/{id}")
    fun getCore(@Path("id") id: Long): Single<Core>

    @GET("/crew")
    fun getCrews(): Single<List<Crew>>

    @GET("/crew/{id}")
    fun getCrew(@Path("id") id: Long): Single<Crew>

    @GET("/dragons")
    fun getDragons(): Single<List<Dragon>>

    @GET("/dragons/{id}")
    fun getDragon(@Path("id") id: Long): Single<Dragon>

    @GET("/landpads")
    fun getLandpads(): Single<List<Landpad>>

    @GET("/landpads/{id}")
    fun getLandpad(@Path("id") id: Long): Single<Landpad>

    @GET("/launches/past")
    fun getPastLaunches(): Single<List<Launch>>

    @GET("/launches/upcoming")
    fun getUpcomingLaunches(): Single<Launch>

    @GET("/launches/latest")
    fun getLatestLaunch(): Single<Launch>

    @GET("/launches/next")
    fun getNextLaunch(): Single<Launch>

    @GET("/launches")
    fun getAllLaunches(): Single<List<Launch>>

    @GET("/launches/{id}")
    fun getLaunch(@Path("id") id: Long): Single<Landpad>

    @GET("/launchpads")
    fun getLaunchPads(): Single<List<Launchpad>>

    @GET("/launchpads/{id}")
    fun getLaunchPad(@Path("id") id: Long): Single<Launchpad>

    @GET("/payloads")
    fun getPayLoads(): Single<List<Payload>>

    @GET("/payloads/{id}")
    fun getPayLoad(@Path("id") id: Long): Single<Payload>

    @GET("/roadster")
    fun getRoadster(): Single<List<Roadster>>

    @GET("/rockets")
    fun getRockets(): Single<List<Rocket>>

    @GET("/rockets/{id}")
    fun getRocket(@Path("id") id: Long): Single<Rocket>

    @GET("/ships")
    fun getShips(): Single<List<Ship>>

    @GET("/ships/{id}")
    fun getShip(@Path("id") id: Long): Single<Ship>

    @GET("/starlink")
    fun getStarlinks(): Single<List<Starlink>>

    @GET("/starlink/{id}")
    fun getStarlink(@Path("id") id: Long): Single<Starlink>
}